src/libraries/Microsoft.Extensions.Configuration.Binder/gen/TypeSpecKind.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    internal enum TypeSpecKind
+    {
+        StringBasedParse = 0,
+        Enum = 1,
+        Object = 2,
+        Array = 3,
+        Enumerable = 4,
+        Dictionary = 5,
+        IConfigurationSection = 6,
```

Comment:
just curious, why can't IConfigurationSection be represented with `Dictionary` and `ByteArray` with `Array`/`Enumerable`?

Reply:
Both types need to be treated differently, so separate values for them makes the code cleaner.